arkouda/numeric.py

```diff
@@ -2182,3 +2183,58 @@ def count_nonzero(pda):
         return sum((pda).astype(np.int64))
     elif pda.dtype == str:
         return sum((pda != "").astype(np.int64))
+
+
+def putmask(pda: pdarray, mask: Union[bool, pdarray], values: pdarray):
+    """
+    Overwrite elements of a pdarray at indices where mask is True
+
+    Parameters
+    ----------
+    pda    : pdarray, source data, also output data
+        pda = input where mask is False, = values where mask is True
+    mask   : a scalar boolean, or a pdarray of booleans
+    values : pdarray, replacement data
+
+    Returns
+    -------
+    None - pda is modified in-place
+
+    Notes
+    -----
+    If values.size != a.size, values is repeated and/or pruned as needed to
+    make sizes match, because ak.where requires matching sizes.
+
+    Examples
+    -------
+    >>> a = ak.array(np.arange(10))
+    >>> ak.putmask (a,a>2,a**2)
+    array ([0,1,2,9,16,25,36,49,64,81])
+
+    >>> values = ak.array([3,2])
+    >>> ak.putmask (a,a>2,values)
+    array ([0,1,2,2,3,2,3,2,3,2])
+
+    Raises
+    ------
+    TypeError
+        Raised if a and values are not the same type
+
+    """
+
+    from arkouda.pdarraysetops import concatenate
+
+    # check for matching types
+
+    if values.dtype != pda.dtype:
+        raise TypeError("ak.putmask requires arrays of matching type")
+
+    # if values is not the same size as pda, repeat it and/or prune it as needed
+
+    growth = pda.size // values.size + (0 if pda.size % values.size == 0 else 1)
```

Comment:
bools can be interpreted as ints, so 
```python
+ (0 if pda.size % values.size == 0 else 1)
# is equivalent to
+ (pda.size % values.size != 0)
```

Reply:
I'm inclined to resist that change.  While it's true that these are equivalent, I don't think the meaning of "+ (pda.size % values.size !=0) is as clear.

Reply:
I would argue that adding the boolean rather than the explicit `1/0` is more pythonic and more closely resembles the style of the project as a whole. But ultimately this is just a matter of taste, so im totally down with leaving it as is!